src/matrix/net/common.ts

```diff
@@ -27,9 +35,10 @@ export function encodeQueryParams(queryParams) {
         .join("&");
 }
 
-export function encodeBody(body) {
-    if (body.nativeBlob && body.mimeType) {
-        const blob = body;
+export function encodeBody(body: Record<string, any>): IEncodedBody {
```

Comment:
The problem with `any` is that it also allows primitive values (number, boolean ...) to be passed. I think what we want here might be `BlobHandle | object` since we don't access any keys in body?